plugin/storage/es/spanstore/reader.go

```diff
@@ -90,26 +94,39 @@ type SpanReader struct {
 	serviceOperationStorage *ServiceOperationStorage
 	spanIndexPrefix         string
 	serviceIndexPrefix      string
+	spanConverter           dbmodel.ToDomain
+}
+
+// SpanReaderParams holds constructor params for NewSpanReader
+type SpanReaderParams struct {
+	Client                  es.Client
+	Logger                  *zap.Logger
+	MaxLookback             time.Duration
+	MetricsFactory          metrics.Factory
+	serviceOperationStorage *ServiceOperationStorage
+	IndexPrefix             string
+	TagDeDotChar            string
 }
 
 // NewSpanReader returns a new SpanReader with a metrics.
-func NewSpanReader(client es.Client, logger *zap.Logger, maxLookback time.Duration, metricsFactory metrics.Factory, indexPrefix string) spanstore.Reader {
-	return storageMetrics.NewReadMetricsDecorator(newSpanReader(client, logger, maxLookback, indexPrefix), metricsFactory)
+func NewSpanReader(p SpanReaderParams) spanstore.Reader {
```

Comment:
Is this pattern being used elsewhere (passing a "params" object to a constructor)?

Reply:
For example here https://github.com/jaegertracing/jaeger/blob/master/cmd/ingester/app/consumer/consumer.go#L54. Search on params returns more examples